extensions/stackdriver/metric/record.cc

```diff
@@ -60,54 +56,185 @@ void record(bool is_outbound, const ::Wasm::Common::FlatNode& local_node_info,
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationServiceNamespaceKey(),
+          flatbuffers::GetString(local_node_info.namespace_()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationPortKey(), std::to_string(request_info.destination_port))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourcePrincipalKey(), std::string(request_info.source_principal))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceWorkloadNameKey(),
+          flatbuffers::GetString(peer_node_info.workload_name()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceWorkloadNamespaceKey(),
+          flatbuffers::GetString(peer_node_info.namespace_()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceOwnerKey(), Common::getOwner(peer_node_info))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationPrincipalKey(),
+          std::string(request_info.destination_principal))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationWorkloadNameKey(),
+          flatbuffers::GetString(local_node_info.workload_name()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationWorkloadNamespaceKey(),
+          flatbuffers::GetString(local_node_info.namespace_()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationOwnerKey(), Common::getOwner(local_node_info))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationCanonicalServiceNameKey(),
+          getLocalCanonicalName(local_node_info))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationCanonicalServiceNamespaceKey(),
+          flatbuffers::GetString(local_node_info.namespace_()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          destinationCanonicalRevisionKey(),
+          getLocalCanonicalRev(local_node_info))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceCanonicalServiceNameKey(),
+          getPeerCanonicalName(peer_node_info))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceCanonicalServiceNamespaceKey(),
+          flatbuffers::GetString(peer_node_info.namespace_()))},
+      {std::make_pair<opencensus::tags::TagKey, std::string>(
+          sourceCanonicalRevisionKey(), getPeerCanonicalRev(peer_node_info))}};
+  return inboundMap;
+}
+
+void addHttpSpecificTags(const ::Wasm::Common::RequestInfo& request_info,
+                         TagKeyValueList& tag_map) {
+  const auto& operation =
+      request_info.request_protocol == ::Wasm::Common::kProtocolGRPC
+          ? request_info.request_url_path
+          : request_info.request_operation;
+  tag_map.push_back(std::make_pair(Metric::requestOperationKey(), operation));
```

Comment:
use emplace_back instead to avoid extra copy.

Reply:
Thanks..Fixed